### ISSUE TITLE
Slicing instead of iter+skip

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust-clippy#10196]
### What it does

Suggests to use normal slicing instead of iter() plus skip().

### Category

pedantic

### Advantage

Avoiding iter and skip shortens the code and probably gives a faster compilation.

### Drawbacks

None.

### Example

```rust
<code>
```Clippy of Nightly rustc 2023-01-10 gives no warnings for this code:

```rust
#![warn(clippy::all)]
#![warn(clippy::nursery)]
#![warn(clippy::pedantic)]
fn main() {
    let items = vec![10, 20, 30, 40];
    let i = 0;
    for x in items.iter().skip(i + 1) {
        println!("{x}");
    }
}
```

I think it should suggest to write instead:

```rust
fn main() {
    let items = vec![10, 20, 30, 40];
    let i = 0;
    for x in &items[i + 1 ..] {
        println!("{x}");
    }
}
```

Could be written as:

```rust
<code>
```

**Comment by leonardo-m** [comment]
(The form to suggest new lints is a mess to use).

**Comment by Alexendoo** [comment]
This would produce a panic if the index is out of the slice bounds

**Comment by Jarcho** [comment]
The correct suggestion would be `items.get(i+1..).unwrap_or(&[])`.

This seems to be worth it as a perf lint. The following gets vectorized:

```rust
pub fn square(list: &[u32]) -> u32 {
    let mut sum = 0u32;
    for x in list.get(5..).unwrap_or(&[]) {
        sum = sum.wrapping_add(*x);
    }
    sum
}
```

whereas the version with skip does not:

```rust
pub fn square(list: &[u32]) -> u32 {
    let mut sum = 0u32;
    for x in list.iter().skip(5) {
        sum = sum.wrapping_add(*x);
    }
    sum
}
```

See [godbolt](https://godbolt.org/z/EY5eMs6dY).

**Comment by Jarcho** [comment]
`skip` is now optimized as of 1.70. Since the equivalent suggestion is not simpler I'm going to close this. If this notably affects compilation speed or there are cases which rustc cannot reasonably optimize this can be reconsidered.
